diff --git a/src/C/dense.c b/src/C/dense.c
index 959eaf5..3a2ec66 100644
--- a/src/C/dense.c
+++ b/src/C/dense.c
@@ -158,7 +158,9 @@ matrix * Matrix_New(int nrows, int ncols, int id)
     return NULL;
   }
 
-  a->id = id; a->nrows = nrows; a->ncols = ncols;
+  a->id = id; a->nrows = nrows; a->ncols = ncols; a->ob_exports = 0;
+  a->shape[0] = 0; a->shape[1] = 0;
+  a->strides[0] = 0; a->strides[1] = 0;
   if ((a->buffer = calloc(nrows*ncols,E_SIZE[id])))
     return a;
   else if (nrows*ncols == 0) 
@@ -202,12 +204,17 @@ matrix *Matrix_NewFromPyBuffer(PyObject *obj, int id, int *ndim)
   int flags = PyBUF_FORMAT | PyBUF_STRIDES;  
 
   Py_buffer *view = (Py_buffer *)malloc(sizeof(Py_buffer));
+  if (!view) {
+    PyErr_NoMemory();
+    return NULL;
+  }
   if (PyObject_GetBuffer(obj, view, flags)) {
     free(view);
     PY_ERR_TYPE("buffer not supported"); 
   }
 
   if (view->ndim != 1 && view->ndim != 2) {
+    PyBuffer_Release(view);
     free(view);
     PY_ERR_TYPE("imported array must have 1 or 2 dimensions");
   }
